dev-tools/xtask/src/a4x2/a4x2_deploy.rs

```diff
@@ -0,0 +1,938 @@
+    // automatically download the hotpatched propolis at least. So consider the
+    // comment below and printlns NOT FINAL and something that should be
+    // resolved BEFORE this code is merged.
+    //
+    // This is the code we *would* run, and will, once we have a URL to a
+    // working propolis:
+    // cmd!(sh, "curl")
+    //     .arg("--remote-name")
+    //     .arg("--location")
+    //     .arg("https://buildomat.eng.oxide.computer/wg/0/artefact/01J8TPA802A5SHWPY5K05ZH2J3/ucXqZK2nT2qxxgI0vY5iygOCdEwguBoe9fk3cFrm4IOs9TLX/01J8TPAVRJDHEYKPEG07AHG51H/01J8TQ0QB5M510N99RWD7X55HT/propolis-server")
+    //     .run()?;
+    // cmd!(sh, "chmod +x propolis-server").run()?;
+    // cmd!(sh, "pfexec mv propolis-server /usr/bin").run()?;
+    //
+    // Instead, I am just doing this for now:
+    eprintln!("We cannot currently download the correct propolis for you.");
+    eprintln!("Please retrieve `/staff/mike/propolis-server` from");
+    eprintln!("`catacomb.eng.oxide.computer` and place it in `/usr/bin`.");
+    eprintln!();
+    eprintln!(
+        "scp catacomb.eng.oxide.computer:/staff/mike/propolis-server /tmp/propolis-server"
+    );
+    eprintln!("chmod +x /tmp/propolis-server");
+    eprintln!("pfexec mv /tmp/propolis-server /usr/bin");
+
+    Ok(())
+}
+
+fn unpack_a4x2(
+    sh: &Shell,
+    env: &Environment,
+    tgz_path: &Utf8Path,
+) -> Result<()> {
+    cmd!(sh, "banner 'unpack'").run()?;
+    let a4x2_dir = &env.a4x2_dir;
+
+    // Remove previous execution
+    if a4x2_dir.try_exists()? {
+        fs::remove_dir_all(a4x2_dir)?;
+    }
+
+    if !tgz_path.try_exists()? {
+        bail!(
+            "a4x2-package bundle does not exist at {}, did you run `cargo xtask a4x2 package`?",
+            tgz_path
+        );
+    }
+
+    cmd!(sh, "tar -xvzf {tgz_path}").run()?;
+
+    if !a4x2_dir.try_exists()? {
+        bail!(
+            "extracting a4x2-package bundle did not result in a4x2-package-out/ existing"
+        );
+    }
+
+    // sled-common.tar contains files that were deduplicated (by sha256 hash)
+    // between the cargo bays. A file is deduplicated only if it is present and
+    // the same in *all* bays, so we unconditonally extract to all bays here.
+
+    cmd!(sh, "tar -cf sled-common.tar -C {a4x2_dir}/cargo-bay/sled-common ./")
+        .run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g0").run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g1").run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g2").run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g3").run()?;
+
+    Ok(())
+}
+
+fn prepare_to_launch_a4x2(sh: &Shell, env: &Environment) -> Result<()> {
+    cmd!(sh, "banner 'prepare'").run()?;
+
+    // Generate an ssh key we will use to log into the sleds.
+    // `-N ''` means "empty passphrase"
+    cmd!(sh, "ssh-keygen -t ed25519 -N '' -f a4x2-ssh-key").run()?;
```

Comment:
I'd suggest printing a one line summary of what we did here.  Something like `loaded {n} keys from $HOME/.ssh`